File: Jour_1/02_structure_de_données.js
{
  function allUniq(str) {
    return str.length === new Set(str).size;
  }

  function allUniqWithoutStruc(str) {
    for (let i = 0; i < str.length; i++) {
      for (let j = i + 1; j < str.length; j++) {
        if (str(i) === str(j)) {
          return false;
        }
      }
    }
    return true;
  }
}

console.log(allUniq('abc')); //-> //true)
console.log(allUniq('abca')); //-> //false)

File: Jour_1/02_structure_de_données.py
"""Implement an algorithm to determine if a string has
 all unique characters. What if you cannot use additional 
 data structures?"""

#l'array set n'a pas de doublon par nature
def allUnique(str):
    return str.length === new set(str).size

allUnique("abc")
allUnique("abca")


"""faire une bulle  sort fonction ou un dictionnaire
on doit avoir un return true si pas de répétition et false si 
une lettne re répéte"""